packages/plugin-essentials/sources/commands/whoami.ts

```diff
@@ -0,0 +1,61 @@
+import {Configuration, MessageName, PluginConfiguration, StreamReport} from '@berry/core';
+import {httpUtils}                                                     from '@berry/core';
+import {Clipanion}                                                     from 'clipanion';
+import {Readable, Writable}                                            from 'stream';
+
+// eslint-disable-next-line arca/no-default-export
+export default (clipanion: Clipanion, pluginConfiguration: PluginConfiguration) => clipanion
+
+  .command(`whoami`)
+  .describe(`display username`)
+
+  .detail(`
+    Print the registry username to standard output.
+  `)
+
+  .example(
+    `Print username for the default registry`,
+    `yarn whoami`,
+  )
+
+  .action(async ({cwd, stdin, stdout}: {cwd: string, stdin: Readable, stdout: Writable, registry: string}) => {
+    const configuration = await Configuration.find(cwd, pluginConfiguration);
+
+    const report = await StreamReport.start({configuration, stdout}, async report => {
+      const authorization = getAuthorizationHeader(configuration);
+
+      if (authorization === null)
+        return report.reportError(MessageName.AUTHENTICATION_NOT_FOUND, `Authentication not found`);
+
+      const headers = { authorization };
+
+      try {
+        const registryUrl = getRegistryUrl(configuration);
+        const responseBuffer = await httpUtils.get(`${registryUrl}-/whoami`, configuration, { headers });
```

Comment:
There's a `npmHttpUtils` in `plugin-npm` that handles all the authentication part. I think you can just use it (export it from its `index` if it isn't already the case).

Reply:
#119 also introduces a `getRegistry` function in `npmConfigUtils.ts` (unexported at the moment) for getting the registry for a package

Reply:
I actually exported it (to access it from `publish`) - you can just do this:

```
import {npmConfigUtils} from '@berry/plugin-npm';
```

Reply:
Or actually I just remembered I modified `npmHttpUtils` to automatically deduce the registry based on the `ident` option, so you should just use it instead of `httpUtils` (I don't think `npmHttpUtils` allows `null` as valid ident at the moment, though).